File: solutions/python3/CountTheNumberOfConsistentStrings.py
class Solution:
    def getIntegerBinRepresentation(self, word: str) -> int:
        rep: int = 0
        a = ord('a')
        for ch in word:
            rep |= (1 << (ord(ch) - a))
        return rep


    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed_int: int = self.getIntegerBinRepresentation(allowed)
        ans: int = 0
        temp: int = 0
        for word in words:
            temp = self.getIntegerBinRepresentation(word)
            if temp == temp & allowed_int:
                ans += 1
        return ans
            

File: solutions/java/CountTheNumberOfConsistentStrings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        HashSet<Character> allowedSet = new HashSet();
        for (char c : allowed.toCharArray())
            allowedSet.add(c);

        int res = 0;
        for (String word : words){
            if (isWordConsistent(word, allowedSet))
                res += 1;
        }
        return res;
    }


    public boolean isWordConsistent(String word, HashSet<Character> allowedSet){
        for (char c : word.toCharArray()){
            if (!allowedSet.contains(c))
                return false;
        }
        return true;
    }
}